aws/table_aws_cloudwatch_metric_statistic_data_point.go

```diff
@@ -0,0 +1,173 @@
+	return &plugin.Table{
+		Name:        "aws_cloudwatch_metric_statistic_data_point",
+		Description: "AWS CloudWatch Metric Statistic Data Point",
+		List: &plugin.ListConfig{
+			Hydrate: listCloudWatchMetricStatisticDataPoints,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"InvalidParameterValue"}),
+			},
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "metric_name",
+					Require: plugin.Required,
+				},
+				{
+					Name:    "namespace",
+					Require: plugin.Required,
+				},
+				{
+					Name:      "timestamp",
+					Operators: []string{">", ">=", "=", "<", "<="},
+					Require:   plugin.Required,
+				},
+				{
+					Name:    "period",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "dimensions",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(cloudwatchv1.EndpointsID),
+		Columns: awsRegionalColumns(cwMetricColumns([]*plugin.Column{
+			{
+				Name:        "label",
+				Description: "A label for the specified metric.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "period",
+				Description: "The granularity, in seconds, of the returned data points.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "dimensions",
+				Description: "The dimensions for the metric.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Label"),
+			},
+		})),
+	}
+}
+
+type MetricStatistics struct {
+	MetricName *string
+	Namespace  *string
+	Period     *int32
+	Label      *string
+	Dimensions []types.Dimension
+	types.Datapoint
+}
+
+//// LIST FUNCTION
+
+func listCloudWatchMetricStatisticDataPoints(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// set the input parameters
+	params := &cloudwatch.GetMetricStatisticsInput{
+		MetricName: aws.String(d.EqualsQuals["metric_name"].GetStringValue()),
+		Namespace:  aws.String(d.EqualsQuals["namespace"].GetStringValue()),
+		Statistics: types.Statistic.Values(types.StatisticMaximum),
+	}
+
+	//set the start and end time based on the provided timestamp
+	for _, q := range d.Quals["timestamp"].Quals {
+		time := q.Value.GetTimestampValue().AsTime()
+		switch q.Operator {
+		case "=":
+			params.StartTime = aws.Time(time)
+			params.EndTime = aws.Time(time)
+		case ">=", ">":
+			params.StartTime = aws.Time(time)
+		case "<", "<=":
+			params.EndTime = aws.Time(time)
+		}
+	}
+
+	// set the period based on the duration between the start and end time
+	// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudwatch@v1.25.1#GetMetricStatisticsInput.Period
+	duration := params.EndTime.Sub(*params.StartTime)
+	// if the duration is less than 3 hrs
+	if duration.Hours() < 3 {
```

Comment:
@bigdatasourav Can you please add some comments showing examples of what the period would be and how many API results we can expect for a few different scenarios?

Reply:
updated.